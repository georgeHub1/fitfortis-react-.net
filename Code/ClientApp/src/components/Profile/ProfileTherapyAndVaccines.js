import React, { Component } from 'react';
import { Form } from 'antd';
import { injectIntl} from 'react-intl';
import PropTypes from 'prop-types';
import CustomTag from '../Common/CustomTag';
import CustomCard from '../Common/CustomCard.js';
import styles from './styles.module.less';
import * as R from 'ramda';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';

class ProfileTherapyAndVaccines extends Component {
  static propTypes = {
    setTabValue: PropTypes.func
  };

  static defaultProps = {
    setTabValue: () => true
  };

  constructor (props) {
    super(props);
    this.state = {
      profileTherapyVaccine: props.profileTherapyVaccine
    };
  }

  componentDidMount () {
    analyticId.firebaseAnalyticsLog('Profile_TherapyAndVaccines');
  }
  componentDidUpdate ({profileTherapyVaccine}) {
    const { profileTherapyVaccine: profileTherapyVaccineProps  } = this.props;

    if (!R.equals(profileTherapyVaccine, profileTherapyVaccineProps))
      this.setState({
        profileTherapyVaccine: [...profileTherapyVaccineProps]
      });
  }

  onClickTag = (tag, key) => {
    const { setTabValue } = this.props;
    const { profileTherapyVaccine } = this.state;
    const dataVaccine = R.clone(profileTherapyVaccine);
    const newElem = R.assoc('isSelected', !tag.isSelected)(tag);

    const indexElem = R.indexOf(tag)(dataVaccine);

    dataVaccine.splice(indexElem, 1, newElem);
    this.setState({
      profileTherapyVaccine: [
        ...dataVaccine
      ]
    });
    setTabValue({
      profileTherapyVaccine: [
        ...dataVaccine
      ]
    });
  };


  render () {
    const profileTherapyVaccine = this.state.profileTherapyVaccine !== undefined
    ? R.groupWith((a, b) => a.key === b.key, R.sortBy(R.prop('key'), R.clone(this.state.profileTherapyVaccine)))
    : [];

    return (
      <div>
        {
          profileTherapyVaccine.map((groupVaccine, index) => {
            return (
              <CustomCard key ={index}>
                <div className={styles.chronicConditionsWrapper}>
                  <div className="cardHeaderProfile">
                  { groupVaccine[0].category }
                  </div>
                  <div className="containerTags">
                    <Form.Item>
                      <CustomTag
                        handleClick={this.onClickTag}
                        tags={ groupVaccine }
                        keyTag='profileTherapyVaccine'
                      />
                    </Form.Item>
                  </div>
                </div>
              </CustomCard>
            );
          })
        }
      </div>
    );
  }
}

export default injectIntl(ProfileTherapyAndVaccines);
