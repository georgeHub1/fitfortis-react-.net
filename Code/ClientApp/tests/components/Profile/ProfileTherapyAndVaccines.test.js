import React from 'react';
import renderer from 'react-test-renderer';

import ProfileTherapyAndVaccines from '../../../src/components/Profile/ProfileTherapyAndVaccines.js';

const Container = ProfileTherapyAndVaccines;

describe('ProfileTherapyAndVaccines', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(<Container/>)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
