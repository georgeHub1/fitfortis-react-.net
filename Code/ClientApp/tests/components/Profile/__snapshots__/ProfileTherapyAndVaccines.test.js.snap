// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileTherapyAndVaccines renders correctly 1`] = `<div />`;
